### DISCUSSION TITLE
スコアの計算過程（根拠）をよりわかりやすく表示するためにどうすべきか？（オープンディスカッション）

### DISCUSSION THREAD

**Discussion by nunukim** [lapras-inc/lapras-user-community#25] [スコアについてのご意見・議論]
# モチベーション

スコアv2.0でコードを公開しますが、実際にどのアウトプットがどのように計算に寄与しているかをプロダクト上でも確認できるようにすることで、よりスコアの透明性と納得感が上がるのではないかと考えられます。

しかしながら、計算の途中結果を保持して表示できるようにすると、計算ロジックが表示方法に依存するようになり、ロジックの改修がしづらくなってしまうというトレードオフもあります。

また、スコアの計算式は(EXPとは異なり)単純な和ではないため、個々のアウトプットの寄与を正確に表すのは難しいという問題もあります。

スコアの計算過程を表示するアイディアについてご意見をいただけたらと思います。

**Comment by rockymanobi** [comment]
全部は無理でも GitHub、TechArticleなどは計算対象となった上位3つのアウトプットがどれなのか、が見えるだけでもうれしい気がしています！

> 計算の途中結果を保持して表示できるようにすると、計算ロジックが表示方法に依存するようになり、ロジックの改修がしづらくなってしまうというトレードオフ

でもこれはほんとうそうですね

**Reply by nunukim** [reply]
そうなんですよね。。
「上位3つ」に関しては、 #24 によって変わる可能性が高い気がしており、こちらの改修がある程度落ち着いてきてから考えたいところではあります・・・

**Reply by akinomyoga** [reply]
#24 で言う $`x_n`$ が各リポジトリ・各記事に付記されるだけでも感覚が掴めて良いのではと感じます (それだけでは最終的なスコアに「どう寄与」しているかまでは分からないですが)。
